src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/MappingResource.kt

```diff
@@ -164,4 +168,32 @@ class MappingResource(private val mappingService: MappingService) {
     ]
   )
   suspend fun deleteVisitIdMappings() = mappingService.deleteVisitMappings()
+
+  @PreAuthorize("hasRole('ROLE_READ_MAPPING')")
+  @GetMapping("/mapping/migration-id/{migrationId}")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "get paged mappings by migration id",
+    description = "Retrieve all mappings for the given migration id (identifies a single migration run). Results are paged.",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Mapping page returned",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = MappingDto::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  suspend fun getVisitMappingsByMigrationId(
+    @PageableDefault pageRequest: Pageable,
+    @Schema(description = "Migration Id", example = "2020-03-24T12:00:00", required = true)
+    @PathVariable migrationId: String
+  ): Page<MappingDto> {
+    return mappingService.getVisitMappingsByMigrationId(pageRequest = pageRequest, migrationId = migrationId)
+      .awaitSingle()
+  }
```

Comment:
awaitSingle for conversion from flux for the suspend function (can't have a non suspend function unless its a separate resource)

Reply:
there might be an alternative to awaitSingle ...

Reply:
extremely minor, but might be slightly better for the awaitSingle to move down to the service